### PR TITLE
Fix settings on socket.io to use defaults and reuse the existing connection

### PR DESCRIPTION
**Pull request by ruffrey**
MER-4413

1. We should honor the connection timeout that the server sets. This was probably getting out of sync where the client was disconnecting or thought it was disconnected, so created a new connection, but the server (may have) kept the connection longer.
2. `sync disconnect on unload` only applies to the browser
3. Reuse the connection object instead of instantiating a new one. I am unsure why we opted to do this but it was almost certainly the source of the crash we've been seeing:

```
Uncaught exception
    TypeError: Cannot call method 'onClose' of null
        at [object Object]._onTimeout (/Users/jeffparrish/apollo/node_modules/respoke-admin/node_modules/socket.io-client/lib/socket.js:281:22)
        at Timer.listOnTimeout [as ontimeout] (timers.js:110:15)
```

I was able to produce the crash before making the changes, and unable to produce it after, even doing many network disconnect and VPN cycles.

### CODE REVIEW

File: lib/client.js [hunk 1]
```diff
@@ -486,11 +485,11 @@ function Client(options) {
             'App-Token': self.tokens['App-Token'],
             endpointId: null,
             connectParams: {
-                'connect timeout': 2000,
+                // 'connect timeout': 2000,
```
Comment (leedm777) [inline lib/client.js]: Remove instead of commenting out.

**Comment by leedm777** [comment]
> Reuse the connection object instead of instantiating a new one. I am unsure why we opted to do this but it was almost certainly the source of the crash we've been seeing.

We do this in Respoke.js, because to reconnect we usually have to get a new auth-token. Since getting an auth-token is an application level thing, the application needs to control the reconnects.

**Comment by ruffrey** [comment]
well, these are failing the tests. whoopsy daisy

**Comment by leedm777** [comment]
@ruffrey re: failing tests, I think that's because there's encrypted data (the app secret) that the fork can't access.

**Comment by leedm777** [comment]
> We should honor the connection timeout that the server sets. 

Does that even work for connection timeouts? If the client can't connect to the server, how will it know what the server's connection timeout is?

FWIW, we upped this to 10 seconds in Respoke.js, and made it configurable.

**Comment by ruffrey** [comment]
Good point re: connection timeouts. It should use the socket.io default though.

**Comment by ruffrey** [comment]
Ok we are back to a good build.

The issues was...manually setting the socket timeout instead of deferring to the respoke api. No problem with socket.io.

There were a ton of problems with shared state in the tests, which should be fixed.

Also the method `respoke.close()`, when called on a client without an active socket, was crashing.

Finally I took 5 mins to update the docs page to look more like docs.respoke.io

**Comment by ruffrey** [comment]
image

**Comment by chadxz** [comment]
i4zrp6t

:+1:

**Comment by dpirestani** [comment]
Jeff

We haven't met, but I have been following your work closely.

I am looking for a good SignalR architect, would you happen to know anyone?

I founded a company called InDemand Interpreting video remote interpreting services www.indemandinterpreting.com<http://www.indemandinterpreting.com> .

Best

Daniel Pirestani


On Nov 3, 2015, at 1:27 PM, Jeff H. Parrish <notifications@github.com<mailto:notifications@github.com>> wrote:


Ok we are back to a good build.

The issues was...manually setting the socket timeout instead of deferring to the respoke api. No problem with socket.io<http://socket.io>.

There were a ton of problems with shared state in the tests, which should be fixed.

Also the method respoke.close(), when called on a client without an active socket, was crashing.

Finally I took 5 mins to update the docs page to look more like docs.respoke.io<http://docs.respoke.io>

-

**Comment by ruffrey** [comment]
apparently that double resolve error was actually fixing something? now the tests fail the same on travis and my local dev

ayy-lmao

**Comment by ruffrey** [comment]
closing since this got messy

**Pull request closed without merging**